wasi/wasi.go

```diff
@@ -52,12 +53,13 @@ type API interface {
 	// See ArgsSizesGet
 	// See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#args_get
 	// See https://en.wikipedia.org/wiki/Null-terminated_string
-	ArgsGet(ctx *wasm.HostFunctionCallContext, argv, argvBuf uint32) Errno
+	ArgsGet(ctx api.HostFunctionCallContext, argv, argvBuf uint32) Errno
 
 	// ArgsSizesGet is a WASI function that reads command-line argument data (Args) sizes.
 	//
-	// There are two result parameters: these are offsets in the wasm.HostFunctionCallContext Memory Buffer to write
-	// corresponding sizes in uint32 little-endian encoding.
+	// There are two result parameters: these are offsets in the api.HostFunctionCallContext Memory to write
+	// corresponding sizes in uint32 little-endian encoding. If either are invalid due to memory constraints, this
```

Comment:
fyi @r8d8, this is the sort of comment I'm adding about errno choices. you can make one similarly, and if we need to change later we can update 

> If either are invalid due to memory constraints, this returns ErrnoFault.